comm/src/main/scala/coop/rchain/comm/transport/TcpServerObservable.scala

```diff
@@ -44,18 +48,21 @@ class TcpServerObservable(
       def send(request: TLRequest): Task[TLResponse] =
         request.protocol
           .fold(internalServerError("protocol not available in request").pure[Task]) { protocol =>
-            Task.delay(bufferTell.pushNext(Send(protocol))).map {
-              case false => internalServerError("message dropped")
-              case true  => noResponse
-            }
+            rPConfAsk.reader(_.local) >>= (
+                src =>
+                  Task.delay(bufferTell.pushNext(Send(protocol))).map {
+                    case false => internalServerError("message dropped")
```

Comment:
you only need `src` when `case true`, this should be rather sth like

```
Task.delay(bufferTell.pushNext(Send(protocol))) >>= {
     case false => internalServerError("message dropped").pure[Task]
     case true => rPConfAsk.reader(_.local).map(noResponse)
```

Reply:
@rabbitonweb `NoResponse` has a `header` but we never sent one. Since we removed roundtrip all response messages had no headers and were ignored them in the `SslSessionClientInterceptor`. On the other hand, we need to cope with a (potential) dynamic IP address of the local node. That's why we need to ask `RPConf` every time we want to use the `local` node.